controllers/snapshot/snapshot_adapter.go

```diff
@@ -427,16 +433,7 @@ func (a *Adapter) getExistingEnvironmentForScenario(integrationTestScenario *v1b
 }
 
 // createEnvironmentForScenario creates new epehemeral environment for scenario by copying existing environment
-func (a *Adapter) createEnvironmentForScenario(integrationTestScenario *v1beta1.IntegrationTestScenario) (*applicationapiv1alpha1.Environment, error) {
-	//get the existing environment according to environment name from integrationTestScenario
-	existingEnv, err := a.getEnvironmentFromIntegrationTestScenario(integrationTestScenario)
-	if err != nil {
-		a.logger.Error(err, "Failed to find the env defined in integrationTestScenario",
-			"integrationTestScenario.Namespace", integrationTestScenario.Namespace,
-			"integrationTestScenario.Name", integrationTestScenario.Name)
-		return nil, err
-	}
-
+func (a *Adapter) createEnvironmentForScenario(existingEnv *applicationapiv1alpha1.Environment, integrationTestScenario *v1beta1.IntegrationTestScenario) (*applicationapiv1alpha1.Environment, error) {
```

Comment:
Given the change in complexity, I think that we can skip having this function altogether and directly use `a.createCopyOfExistingEnvironment()` [here](https://github.com/redhat-appstudio/integration-service/pull/381/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R486) instead of it.